frontend/src/components/pages/AdminDashboard/EditCheckInOrFoodRescue.tsx

```diff
@@ -0,0 +1,125 @@
+import * as Routes from "../../../constants/Routes";
+import { Content } from "../../../types/ContentTypes";
+import SaveButton from "../Scheduling/SaveChangesButton";
+
+const EditCheckInOrFoodRescue = ({
+  isCheckInView = false,
+}: {
+  isCheckInView?: boolean;
+}): React.ReactElement => {
+  const [content, setContent] = useState<Content>();
+  const [{ description, url }, setValue] = useForm({
+    description: "",
+    url: "",
+  });
+
+  const history = useHistory();
+  const navigateToViewPage = () => {
+    history.push(
+      isCheckInView ? Routes.ADMIN_CHECK_INS : Routes.ADMIN_VIEW_DONATIONS,
+    );
+  };
+
+  useEffect(() => {
+    const getContent = async () => {
+      const contentResponse = await ContentAPIClient.getContent();
+      setContent(contentResponse);
+    };
+
+    getContent();
+  }, []);
+
+  const onSaveClick = async () => {
+    const contentResponse = await ContentAPIClient.updateContent(
+      content?.id ?? "",
+      {
+        checkinDescription:
+          (isCheckInView ? description : content?.checkinDescription) ?? "",
+        checkinUrl: (isCheckInView ? url : content?.checkinUrl) ?? "",
+        foodRescueDescription:
+          (!isCheckInView ? description : content?.foodRescueDescription) ?? "",
+        foodRescueUrl: (!isCheckInView ? url : content?.foodRescueUrl) ?? "",
+      },
+    );
+
+    setContent(contentResponse);
+    navigateToViewPage();
+  };
+
+  useEffect(() => {
+    const getContent = async () => {
+      const contentResponse = await ContentAPIClient.getContent();
+      setContent(contentResponse);
+    };
+
+    getContent();
+  }, []);
+
+  return (
+    <Container variant="responsiveContainer" maxWidth="100%" mt="2rem">
+      <Box>
+        <Box display="flex" justifyContent="right">
+          <CloseButton onClick={navigateToViewPage} />
+        </Box>
+        <Text textStyle="desktopHeader4" color="black.100">{`Edit ${
+          isCheckInView ? `check in` : `food rescue`
+        } description`}</Text>
+
+        <Text textStyle="desktopSubtitle" color="black.100" mt="2rem">
+          Edit description
+        </Text>
+        <Textarea
+          mt="2.5rem"
+          value={description}
+          placeholder={
+            isCheckInView
+              ? content?.checkinDescription
+              : content?.foodRescueDescription
+          }
+          name="description"
+          onChange={setValue}
+          width="478px"
+          height="210px"
+          background="squash.100"
+          p="1.5rem"
+        />
+
+        <Text textStyle="desktopSubtitle" color="black.100" mt="3.5rem">
+          Edit link
+        </Text>
+
+        <Input
+          mt="2.5rem"
+          value={url}
+          name="url"
+          placeholder={
+            isCheckInView ? content?.checkinUrl : content?.foodRescueUrl
+          }
+          onChange={setValue}
+          width="44rem"
+          height="64px"
```

Comment:
styling is off on mobile for the Input here
![image](https://user-images.githubusercontent.com/19617248/161658375-595596ef-cc8d-42fc-8f07-641c91e523f3.png)